.github/workflows/conda-pytest.yaml

```diff
@@ -5,32 +5,44 @@ on:
     types: [opened, reopened]
 
 jobs:
-  push_to_registry:
-    name: Build docker file for testing
+  test-unit:
     runs-on: ubuntu-latest
+    container: quay.io/condaforge/miniforge3:latest
+
     steps:
       - name: Check out the repo
         uses: actions/checkout@v2
 
-      - uses: conda-incubator/setup-miniconda@v2
+      - name: Setup uv
+        uses: astral-sh/setup-uv@v3
+        with:
+          enable-cache: true
+          cache-dependency-glob: "pyproject.toml"
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version-file: "pyproject.toml"
+          python-version: '3.12' 
+      
+      - name: Install ESMF
```

Comment:
`Install ESMT and gdal` (im not sure what it is, but im sure its needed)

Reply:
[`f768ab7` (#192)](https://github.com/openclimatefix/uk-pvnet-app/pull/192/commits/f768ab7f6c898322d64a4351c8d9a42950b4ae4f)